lib/src/screens/nano/nano_change_rep_page.dart

```diff
@@ -74,8 +97,20 @@ class NanoChangeRepPage extends BasePage {
 
                             if (confirmed) {
                               try {
-                                await nano!.changeRep(_wallet, _addressController.text);
-                                Navigator.of(context).pop();
+                                _settingsStore.defaultNanoRep = _addressController.text;
+                                // we can only submit a change block if we have an existing transaction history:
+                                if (_wallet.transactionHistory.transactions.isNotEmpty) {
+                                  await nano!.changeRep(_wallet, _addressController.text);
+                                }
```

Comment:
it's better to have any logic in the business logic layer not in the UI/presentation layer, so that if `nano!.changeRep` is used in anywhere else for any reason, we don't have to worry about any unexpected behavior.

i.e
you should call the `await nano!.changeRep(_wallet, _addressController.text);` normally and add the check for transactionHistory inside the `changeRep` function in `lib/nano/cw_nano.dart`